runtime_scan/pkg/orchestrator/discovery/scope.go

```diff
@@ -65,3 +64,74 @@ func (sd *ScopeDiscoverer) Start(ctx context.Context) {
 		}
 	}()
 }
+
+func (sd *ScopeDiscoverer) DiscoverAndCreateAssets(ctx context.Context) error {
+	discoveryTime := time.Now()
+
+	assetTypes, err := sd.providerClient.DiscoverTargets(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to discover assets from provider: %w", err)
+	}
+
+	errs := []error{}
+	for _, assetType := range assetTypes {
+		targetData := models.Target{
+			TargetInfo: utils.PointerTo(assetType),
+			LastSeen:   &discoveryTime,
+		}
+		_, err := sd.backendClient.PostTarget(ctx, targetData)
+		if err == nil {
+			continue
+		}
+
+		var conflictError backendclient.TargetConflictError
+		if !errors.As(err, &conflictError) {
+			// If there is an error, and its not a conflict telling
```

Comment:
typos the IDE alerted on.
```
			// If there is an error, and it's not a conflict telling
			// us that the target already exists, then we need to
			// keep track of it and log it as a failure to
			// complete discovery. We don't fail instantly here
			// because discovering the assets is a heavy operation,
			// so we want to give the best chance to create all the
			// assets in the DB before failing.
```